docs/source/bouncing.rst

```diff
@@ -0,0 +1,136 @@
+How PaaSTA Bounces New Code and SOA-config changes
+==================================================
+
+In the context of this document, "Bouncing" refers to the procedure of
+replacing old tasks with new ones. With long-running services bouncing
+is implemented by controlling Marathon Apps. For periodic tasks, see the
+section on "Chronos Bouncing."
+
+A "Bounce" can happen for one of these reasons:
+
+* A new version of the code is deployed (a git sha change)
+* A change of `soa-configs <yelpsoa_configs.html>`_ for a service. (Change in ram, cpu, environment variables, etc)
+* An issue of a ``paasta restart`` (or partially from a start/stop)
+* A change in system-wide PaaSTA configuration (defaults for volumes, ram, cpu, etc)
+
+By default, PaaSTA will do the safest thing possible and favors service uptime
+over speed. Read further for the actual details of this default procedure in the
+``crossover`` bounce.
+
+If service uptime is irrelevant then the ``brutal`` bounce can be selected.
+
+The ``upthendown`` and ``downthenup`` provide for other use cases where more
+predictable bouncing behavior is desired.
+
+Read more in the next section for exact details and differences amongst these
+bounce method.
+
+Note that in **no case** will PaaSTA "revert code or config" back to previous
+versions after a failed bounce. While some bounce methods have some protection
+against failure situations, PaaSTA will never undo the intent implied by a change.
+Addressing this problem should be done at at a layer *above PaaSTA*.
+
+
+Bounce Methods
+^^^^^^^^^^^^^^
+
+crossover
+"""""""""
+
+The ``crossover`` bounce is the default bounce in PaaSTA. It is the safest
+bounce method, as it only removes old tasks after verifying that the new tasks
+are alive and healthy. The ``crossover`` bounce can actually protect
+infrastructure against bad code, because it won't even proceed if the new
+copies of the code are not healthy!
+
+The cost of the ``crossover`` bounce is time. It takes time for new task to be
+healthy, and the ``crossover`` bounce will take as much time as it takes for
+the new copies of the service to start up and become healthy, plus the time it
+takes to drain and kill the old tasks. In the case of HTTP services using
+SmartStack, **no web requests should be lost under normal circumstances**.
+
+.. image:: bounce_crossover.png
+
+
+brutal
+""""""
+
+The ``brutal`` bounce replaces old tasks with new tasks with no regards to
+safety. It launches new tasks immediately and kills old tasks immediately
+without draining them.
+
+The ``brutal`` bounce is probably best for apps that don't have safety concerns
+and want the fastest bounce possible.
+
+.. image:: bounce_brutal.png
+
+**Note1**: Services *will* incur downtime during the ``brutal`` bounce and
```

Comment:
Do we want to mention any hints/tips for adjusting the monitoring settings to avoid getting notified for every bounce?